app/calendar/event-creator.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+import { useState } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+export default function EventCreator({ username, onEventCreated }) {
+  const [name, setName] = useState("");
+  const [date, setDate] = useState("");
+  const [calendarId, setCalendarId] = useState("");
+
+  function validateInput() {
+    var valid = true;
+    var errstyle = "2px solid red";
+    var initstyle = "1px solid grey";
+
+    if (name.length == 0) {
+      const el = document.getElementById("name");
+      el.style.border = errstyle;
+      valid = false;
+      el.onchange = () => {
+        (el.style.border = initstyle), (name) => setName(name);
+      };
+    }
+
+    if (date == null || date.length == 0) {
+      const el = document.getElementById("date");
+      el.style.border = errstyle;
+      valid = false;
+      el.onclick = () => {
+        (el.style.border = initstyle), (date) => setDate(date);
+      };
+    }
+
+    if (calendarId.length == 0) {
+      const el = document.getElementById("calendar");
+      el.style.border = errstyle;
+      valid = false;
+      el.onchange = () => {
+        (el.style.border = initstyle),
+          (calendarId) => setCalendarId(calendarId);
+      };
+    }
+    valid ? handleClick() : alert("Invalid Input");
+  }
+
+  function handleClick() {
+    fetch("http://localhost:8080/event", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        username,
```

Comment:
I send the username to the /event endpoint in order to know to which user this event belongs